### PR TITLE
Windows and Joins work when columns have alias

### PR DESCRIPTION
**Pull request by amytai**
Also make sure that columns in the DDlog output relation appear in the same order as in the original SQL create statement.

Signed-off-by: Amy Tai <amy.tai.2009@gmail.com>

### CODE REVIEW

File: sql/src/test/java/ddlog/WindowTest.java [hunk 1]
```diff
@@ -381,4 +381,93 @@ public void joinWindowTest() {
...
+                "(var avg9 = (64'sd0, 64'sd0): (signed<64>, signed<64>));\n" +
+                "(for ((i, _) in g) {\n" +
+                "var v7 = i;\n" +
+                "(var incr = v7.tmp);\n" +
+                "(avg9 = agg_avg_signed_R(avg9, incr))}\n" +
+                ");\n" +
+                "(Tagg{.avg = avg_signed_R(avg9)})\n" +
+                "}\n" +
+                "\n" +
+                "input relation Rt1[Tt1]\n" +
```
Comment (mbudiu-vmw) [inline sql/src/test/java/ddlog/WindowTest.java:407]: usually I write `this.relations(false)`; it's just slightly more robust if you add more input relations.

File: sql/src/main/java/com/vmware/ddlog/translator/TranslationVisitor.java [hunk 2]
```diff
@@ -1040,11 +1040,50 @@ private DDlogIRNode processWindows(QuerySpecification query,
...
+                }
+            } else if (originalColumn instanceof SingleColumn) {
+                SingleColumn scOrigCol = (SingleColumn) originalColumn;
+                boolean found = false;
+                for (SingleColumn sc: Utilities.concatenate(aggregateItems, windowItems)) {
+                    // Aggregate or window items should be referenced by their name in the OverInput or Over tables
+                    // created by this Visitor.
+                    if (scOrigCol.getExpression().equals(sc.getExpression())) {
+                        Expression repl = subst.rewrite(sc.getExpression(), null);
+                        Expression repl1 = dropTable.rewrite(repl, null);
```
Comment (mbudiu-vmw) [inline sql/src/main/java/com/vmware/ddlog/translator/TranslationVisitor.java:1063]: maybe we should rename the 'dropTable' variable as 'dropTableSpecRewriter' or something like that.

**Review by mbudiu-vmw — CHANGES_REQUESTED** [review]

File: sql/src/test/java/ddlog/WindowTest.java (hunk 1 above)
Comment (amytai) [reply sql/src/test/java/ddlog/WindowTest.java:407]: done

File: sql/src/main/java/com/vmware/ddlog/translator/TranslationVisitor.java (hunk 2 above)
Comment (amytai) [reply sql/src/main/java/com/vmware/ddlog/translator/TranslationVisitor.java:1063]: How about `dropTableExprRewriter`?

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**